pkg/test/integration/common/framework.go

```diff
@@ -0,0 +1,815 @@
+	if len(mcContainerImageTag) != 0 && len(mcmContainerImageTag) != 0 {
+
+		/* if any of mcmContainerImage  or mcContainerImageTag flag is non-empty then,
+		create/update machinecontrollermanager deployment in the control-cluster with specified image
+		*/
+		if c.ControlCluster.IsSeed(c.TargetCluster) {
+			ginkgo.By("Updating MCM Deployemnt")
+			gomega.Expect(c.prepareMcmDeployment(mcContainerImageTag, mcmContainerImageTag, false)).To(gomega.BeNil())
+		} else {
+			ginkgo.By("Creating MCM Deployemnt")
+			gomega.Expect(c.prepareMcmDeployment(mcContainerImageTag, mcmContainerImageTag, true)).To(gomega.BeNil())
+		}
+
+	} else {
+		/*
+		 run mc and mcm locally
+		*/
+		if c.ControlCluster.IsSeed(c.TargetCluster) {
+			ginkgo.By("Scaledown existing machine controllers")
+			gomega.Expect(c.scaleMcmDeployment(0)).To(gomega.BeNil())
+		}
+
+		ginkgo.By("Starting Machine Controller ")
+		args := strings.Fields(fmt.Sprintf("make --directory=%s start CONTROL_KUBECONFIG=%s TARGET_KUBECONFIG=%s CONTROL_NAMESPACE=%s LEADER_ELECT=false ",
+			"../../..",
+			c.ControlCluster.KubeConfigFilePath,
+			c.TargetCluster.KubeConfigFilePath,
+			controlClusterNamespace))
+		outputFile, err := helpers.RotateLogFile(mcLogFile)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		sess, err := gexec.Start(exec.Command(args[0], args[1:]...), outputFile, outputFile)
+		mcsession = sess
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		gomega.Expect(mcsession.ExitCode()).Should(gomega.Equal(-1))
+
+		ginkgo.By("Starting Machine Controller Manager")
+		args = strings.Fields(fmt.Sprintf("make --directory=%s start CONTROL_KUBECONFIG=%s TARGET_KUBECONFIG=%s CONTROL_NAMESPACE=%s LEADER_ELECT=false MACHINE_SAFETY_OVERSHOOTING_PERIOD=300ms",
+			mcmRepoPath,
+			c.ControlCluster.KubeConfigFilePath,
+			c.TargetCluster.KubeConfigFilePath,
+			controlClusterNamespace))
+		outputFile, err = helpers.RotateLogFile(mcmLogFile)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		mcmsession, err = gexec.Start(exec.Command(args[0], args[1:]...), outputFile, outputFile)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		gomega.Expect(mcmsession.ExitCode()).Should(gomega.Equal(-1))
+	}
+
+	ginkgo.By("Setup MachineClass")
+	gomega.Expect(c.setupMachineClass()).To(gomega.BeNil())
+
+	// initialize orphan resource tracker
+	ginkgo.By("Looking for machineclass resource in the control cluster")
+	machineClass, err := c.ControlCluster.McmClient.MachineV1alpha1().MachineClasses(controlClusterNamespace).Get(testMachineClassResources[0], metav1.GetOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By("Determining control cluster name")
+	clusterName, err := c.ControlCluster.ClusterName()
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By("Looking for secrets refered in machineclass in the control cluster")
+	secretData, err := c.ControlCluster.GetSecretData(machineClass.Name, machineClass.SecretRef, machineClass.CredentialsSecretRef)
```

Comment:
This is rightly done! Just a small change that would be required is to return the `*"k8s.io/api/core/v1".Secret` Type from method `GetSecretData` instead of `map[string]string`. Because with `map`, the upcoming call of `c.resourcesTracker.InitializeResourcesTracker` fails.

```
# github.com/gardener/machine-controller-manager-provider-aws/test/integration/controller_test [github.com/gardener/machine-controller-manager-provider-aws/test/integration/controller.test]
./controller_test.go:39:50: cannot use &provider.ResourcesTrackerImpl{} (type *provider.ResourcesTrackerImpl) as type helpers.ResourcesTrackerInterface in argument to common.NewIntegrationTestFramework:
	*provider.ResourcesTrackerImpl does not implement helpers.ResourcesTrackerInterface (wrong type for InitializeResourcesTracker method)
		have InitializeResourcesTracker(*"github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1".MachineClass, *"k8s.io/api/core/v1".Secret, string) error
		want InitializeResourcesTracker(*"github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1".MachineClass, map[string][]byte, string) error
```

Reply:
Thanks @AxiomSamarth,  please check if revendoring fixes this. I have updated the InitializeResourcesTracker